scripts/annotate_metadata_with_index.py

```diff
@@ -18,6 +18,7 @@
     index = pd.read_csv(
         args.sequence_index,
         sep="\t",
+        dtype={"strain": "string"}
```

Comment:
_(non-blocking)_

Personally, I'd put `"strain"` in a variable like `strain_col`, since the value is now used 3 times in this file. Would be easier to see where it is used, in case there are ever plans to add support for `"name"` [like `augur.io.read_metadata()`](https://github.com/nextstrain/augur/blob/f68c4488de1ba6f7c9404b21dd2958bae552a736/augur/io.py#L47). But this is obviously fine!

Reply:
Ah yeah, we should be supporting both 'strain' and 'name' since that's what we say is allowed in the [docs](https://docs.nextstrain.org/projects/ncov/en/latest/guides/data-prep/local-data.html#required-metadata).

Reply:
Thank you both! The original code crashed when the input metadata used "name" instead of "strain" as a column. c6d6b49 fixes the issue by indexing the sequence index data frame by its hardcoded "strain" column (the only value it ever gets) and using a pandas `join` instead of a `merge` to perform the join on the data frame indices. This allows us to avoid knowing the name of the column that the metadata was indexed by and reduces the number of times we refer to "strain".